pkg/validator/shoot_validator.go

```diff
@@ -87,6 +87,25 @@ func (v *Shoot) validateShoot(valContext *validationContext) field.ErrorList {
 	allErrors = append(allErrors, gcpvalidation.ValidateWorkers(valContext.shoot.Spec.Provider.Workers, workersPath)...)
 	allErrors = append(allErrors, gcpvalidation.ValidateControlPlaneConfig(valContext.controlPlaneConfig, allowedZones, workersZones(valContext.shoot.Spec.Provider.Workers), controlPlaneConfigPath)...)
 
+	// WorkerConfig
+	fldPath := field.NewPath("worker", "wokerconfig")
+	for i, worker := range valContext.shoot.Spec.Provider.Workers {
+		if worker.ProviderConfig != nil {
+			workerConfig, err := decodeWorkerConfig(v.decoder, worker.ProviderConfig, fldPath.Index(i).Child("providerConfig"))
+			if err != nil {
+				allErrors = append(allErrors, field.InternalError(fldPath, err))
```

Comment:
Why internal error? It should probably be `field.Invalid`, or? If it's not decodeable then the user has submitted an invalid config.

Reply:
Accept